wallet-core/src/ffi.rs

```diff
@@ -619,3 +630,187 @@ pub unsafe fn moonlight_to_phoenix(
 
     ErrorCode::Ok
 }
+
+#[no_mangle]
+pub unsafe fn moonlight_stake(
+    seed: &Seed,
+    sender_index: u8,
+    stake_value: *const u64,
+    gas_limit: *const u64,
+    gas_price: *const u64,
+    nonce: *const u64,
```

Comment:
I actually disagree with that, since I want to keep the common arguments across all the functions, instead of changing name: since it's a `moonlight` transaction, the `nonce` is the moonlight one. The other one is clearly indicated by the prefix `stake_nonce`. I would understand the confusion if it was an array of nonces.